apps/roam/src/components/DiscourseNodeSearchMenu.tsx

```diff
@@ -0,0 +1,420 @@
+    if (match) {
+      setSearchTerm(match[1]);
+    } else {
+      onClose();
+      return;
+    }
+  }, [textarea, onClose, setSearchTerm, triggerPosition]);
+
+  const keydownListener = useCallback(
+    (e: KeyboardEvent) => {
+      if (e.key === "ArrowDown") {
+        setActiveIndex((prev) => (prev + 1) % allItems.length);
+        e.preventDefault();
+        e.stopPropagation();
+      } else if (e.key === "ArrowUp") {
+        setActiveIndex(
+          (prev) => (prev - 1 + allItems.length) % allItems.length,
+        );
+        e.preventDefault();
+        e.stopPropagation();
+      } else if (e.key == "ArrowLeft" || e.key === "ArrowRight") {
+        e.stopPropagation();
+        e.preventDefault();
+      } else if (e.key === "Enter") {
+        if (allItems.length > 0) {
+          e.preventDefault();
+          e.stopPropagation();
+          onSelect(allItems[activeIndex].item);
+        }
+      } else if (e.key === "Escape") {
+        onClose();
+        e.preventDefault();
+        e.stopPropagation();
+      }
+    },
+    [allItems, setActiveIndex, onSelect, onClose],
+  );
+
+  useEffect(() => {
+    if (activeIndex >= allItems.length) {
+      setActiveIndex(0);
+    }
+  }, [allItems, activeIndex]);
+
+  useEffect(() => {
+    textarea.addEventListener("input", handleTextAreaInput);
+    return () => {
+      textarea.removeEventListener("input", handleTextAreaInput);
+    };
+  }, [handleTextAreaInput, textarea]);
+
+  useEffect(() => {
+    const listeningEl = !!textarea.closest(".rm-reference-item")
+      ? textarea.parentElement
+      : textarea;
+    listeningEl?.addEventListener("keydown", keydownListener);
+    return () => {
+      listeningEl?.removeEventListener("keydown", keydownListener);
+    };
+  }, [keydownListener]);
+
+  useEffect(() => {
+    setTimeout(() => {
+      handleTextAreaInput();
+    }, 50);
+  }, [handleTextAreaInput]);
+
+  let currentGlobalIndex = -1;
+
+  const handleTypeCheckChange = useCallback(
+    (typeKey: string, e: React.MouseEvent) => {
+      e.preventDefault();
+      e.stopPropagation();
+
+      setCheckedTypes((prev) => ({
+        ...prev,
+        [typeKey]: !prev[typeKey],
+      }));
+
+      setTimeout(() => {
+        textarea.focus();
+        const cursorPos = textarea.selectionStart;
+        textarea.setSelectionRange(cursorPos, cursorPos);
+      }, 0);
+    },
+    [textarea],
+  );
+
+  const handleFilterSectionMouseDown = useCallback((e: React.MouseEvent) => {
+    e.preventDefault();
+    e.stopPropagation();
+  }, []);
+
+  const toggleFilterMenu = useCallback(
+    (e: React.MouseEvent) => {
+      e.preventDefault();
+      e.stopPropagation();
+
+      setIsFilterMenuOpen((prev) => !prev);
+
+      setTimeout(() => {
+        if (textarea) {
+          textarea.focus();
+          const cursorPos = textarea.selectionStart;
+          textarea.setSelectionRange(cursorPos, cursorPos);
+        }
+      }, 50);
```

Comment:
Could you show me what use case this is solving?

Reply:
oh yes, it was to remain in focus, but it the preventDefault()  and stopPropagation() already did the job